src/IRMatch.h

```diff
@@ -1358,6 +1358,7 @@ struct Intrin {
     struct pattern_tag {};
     Call::IntrinsicOp intrin;
     std::tuple<Args...> args;
+    Type optional_type_hint;
```

Comment:
At first I thought this was a type hint for the args and got confused. A comment like this would have helped:
// The type of the output of the intrinsic node. Only necessary in cases where it can't be inferred from the input types (e.g. saturating_cast).

Reply:
Will do!

Reply:
I also updated the `match` method to check for matching the `type_hint` if it is defined - because `saturating_cast(Uint(8), x)` could conceivably match any `saturating_cast` call before this change.